lib/index.js

```diff
@@ -874,41 +887,49 @@ class Image extends RustClass {
 
     let data;
 
-    if (Buffer.isBuffer(src)){
+    if (Buffer.isBuffer(src)) {
       [data, src] = [src, '']
-    } else if (typeof src != 'string'){
+    }
+    else if (typeof src != 'string') {
+      onerror(new Error("'src' property value is neither string nor Buffer type.'"))
       return
-    } else if (/^\s*data:/.test(src)) {
+    }
+    else if (/^\s*data:/.test(src)) {
       // data URI
       let [header, mime, enc] = src.slice(0, 40).match(/^\s*data:(?<mime>[^;]*);(?:charset=)?(?<enc>[^,]*),/) || []
       if (!mime || !enc){
         throw new Error(`Invalid data URI header`)
       }else{
         data = Buffer.from(src.slice(header.length), enc);
```

Comment:
Missed this change which decodes text content (I actually thought it was in the Image changes, but turned out I had added it in the later SVG commit).

```js
      let content = src.slice(header.length)
      if (enc == 'utf8')
        content = decodeURIComponent(content)
      data = Buffer.from(content, enc);
```

Reply:
Do you have a hard time _not_ closing with semicolons? I dunno how you do it. Because it drives me nuts.... my right pinkie is just trained to do that automatically. I've had to go through and remove a ton of them to keep with the current style...   😆   (just funny, no worries)

Reply:
Spend enough years in Python-land and your pinkie begins to atrophy ;)